lab2/devops-js-app.conf

```diff
@@ -0,0 +1,70 @@
+proxy_buffering off;
+
+upstream front_end_app {
+    server localhost:4200;
+}
+
+upstream rest_api_server {
+    server localhost:3000; #provide port of your back-end's app
+}
+
+server {
+    listen 5555;
+    location / {
+        proxy_pass http://front_end_app;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Connection $http_connection;
+        proxy_set_header Host $host;
+    }
+
+    location /api/ {
+        proxy_pass http://rest_api_server/;
+        proxy_http_version 1.1;
+        proxy_set_header Host $host;
+        proxy_set_header X-NginX-Proxy true;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $scheme;
+    }
+}
+
+server {
+    listen 8080;
+    server_name static-app.net;
+    root /Users/Dzianis_Pasiukou1/Documents/projects/shop-angular-cloudfront/dist/app;
+    index index.html;
+    location / {
+        try_files $uri $uri/ /index.html =404;
+    }
+
+    location /api/ {
+        proxy_pass http://rest_api_server/;
+        proxy_http_version 1.1;
+        proxy_set_header Host $host;
+        proxy_set_header X-NginX-Proxy true;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $scheme;
+    }
+}
+
+server {
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+    server_name den-app.net; #your domain name from hosts file
+    ssl_certificate /Users/Dzianis_Pasiukou1/Documents/projects/shop-angular-cloudfront/my-cert.crt;
```

Comment:
not the best place to store certs. consider `/usr/local/etc/nginx/ssl` folder for this